contrib/experimental/great_expectations_experimental/expectations/expect_column_values_are_probabilistically_greater_than_equal_to_threshold.py

```diff
@@ -0,0 +1,170 @@
+"""
+This is a template for creating custom ColumnMapExpectations.
+For detailed instructions on how to use it, please see:
+    https://docs.greatexpectations.io/docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_map_expectations
+"""
+
+import json
+
+import dataprofiler as dp
+import numpy as np
+
+# remove extra tf loggin
+import tensorflow as tf
+
+tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
+
+from great_expectations.execution_engine import (
+    PandasExecutionEngine,
+    SparkDFExecutionEngine,
+    SqlAlchemyExecutionEngine,
+)
+from great_expectations.expectations.expectation import ColumnMapExpectation
+from great_expectations.expectations.metrics import (
+    ColumnMapMetricProvider,
+    column_condition_partial,
+)
+
+
+class ColumnValuesConfidenceToBeGreaterThanOrEqualToThreshold(ColumnMapMetricProvider):
+    """MetricProvider Class for determining if the dataprofiler's model confidence value is\
+        greater than or equal to the user-specified threshold"""
+
+    # This is the id string that will be used to reference your metric.
+    condition_metric_name = (
+        "column_values.prediction_confidence_greater_than_or_equal_to_threshold"
+    )
+
+    condition_value_keys = ("threshold",)
+
+    # This method implements the core logic for the PandasExecutionEngine
+    @column_condition_partial(engine=PandasExecutionEngine)
+    def _pandas(cls, column, threshold, **kwargs):
+        """
+        Implement the yes/no question for the expectation
+        """
+        labeler = dp.DataLabeler(labeler_type="structured")
+        labeler.postprocessor.set_params(is_pred_labels=False)
+        results = labeler.predict(
+            column, predict_options={"show_confidences": True}
+        )
+        return (
+            np.choose(results["pred"].astype(int, copy=False), results["conf"].T)
+            >= threshold
+        )
+
+
+class ExpectColumnValuesToBeGreaterThanOrEqualToThreshold(ColumnMapExpectation):
```

Comment:
To put it in simpler terms, is this Expectation about how effectively DataProfiler can Label a column, irrespective of the specifics of the Label or column contents?

Reply:
correct